### PR TITLE
Native gate options example

### PR DESCRIPTION
**Pull request by irfankhan10**
# Description

Adding a knowledge-article describing usage of a non-default native two-qubit gate (OpType.TK2). Adding 1 *.ipynb accompanied by 3 pngs.

# Related issues
None

# Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented hard-to-understand parts of my code.
- [ ] I have made corresponding changes to the public API documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have updated the changelog with any user-facing changes.

### CODE REVIEW

**Comment by cqc-alec** [comment]
I've removed myself from the reviewers; happy for Megan/Yao to review this.

File: examples/Quantinuum_advanced_compilation.ipynb [hunk 1]
```diff
@@ -0,0 +1,607 @@
...
+  {
+   "cell_type": "markdown",
+   "id": "fc930451",
+   "metadata": {},
+   "source": [
+    "### Decomposing a random $SU(4)$ unitary into a circuit primitive\n",
+    "\n",
+    "To set up the Quantum Volume test, we start by building up the repeated circuit elements. The function in the code cell below defines a `tket` Circuit Box with [`pytket.circuit.CircBox`](https://tket.quantinuum.com/api-docs/circuit.html#pytket.circuit.CircBox). Circuit Boxes are useful for composing larger circuits from smaller subcircuits that utilize the same set of gates.\n",
+    "\n",
+    "The Circuit box below contains a blueprint for the decomposition of a random generalized $SU(4)$ unitary distributed with the Haar Measure into a circuit primitive over 2-qubits. The implementation is based on [arxiv.0609050](http://arxiv.org/abs/math-ph/0609050). The random $SU(4)$ unitary is generated using [`scipy.stats.unitary_group`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.unitary_group.html#scipy-stats-unitary-group)."
```
Comment (yao-cqc) [inline examples/Quantinuum_advanced_compilation.ipynb:237]: Are the generated matrices SU(4) or U(4)?

Comment (irfankhan10) [reply examples/Quantinuum_advanced_compilation.ipynb:237]: They should be SU(4)

Comment (yao-cqc) [reply examples/Quantinuum_advanced_compilation.ipynb:237]:
```python
from scipy.linalg import det
r = unitary_group.rvs(dim=4)
det(r)
```
I've done a quick check and it seems that the determinant is not always 1.

Comment (irfankhan10) [reply examples/Quantinuum_advanced_compilation.ipynb:237]: thanks

Comment (irfankhan10) [reply examples/Quantinuum_advanced_compilation.ipynb]: @yao-cqc I have a made a change to transform U(4) to SU(4) matrix. Please see in the PR

**Review by yao-cqc — DISMISSED** [review]

**Review by mlk621-qtm — DISMISSED** [review]

**Review by yao-cqc — APPROVED** [review]

**Pull request merged**